Slim/Menu/BrowseLibrary/Releases.pm

```diff
@@ -102,7 +102,8 @@ sub _releases {
 	my @items;
 	my $searchTags = [
 		"artist_id:$artistId",
-		"role_id:" . join(',', keys %primaryArtistIds),
+  		# If we've not come from an Artists top level menu (ie we've come from an Additional Browse Modes menu), use the menu role_id to find releases  
+ 		$args->{'params'}->{'menu_mode'} ne 'artists' && $args->{'params'}->{'menu_roles'} ? "role_id:" . $args->{'params'}->{'menu_roles'} : "role_id:" . join(',', keys %primaryArtistIds),
```

Comment:
What would the `menu_mode` typically be here? Looking at the code in `ExtendedBrowseModes` I'm wondering whether it's even defined?

And I believe this code should move up to where we define the `%primaryArtistIds`. Otherwise the grouping would still not happen, would it?

Reply:
`menu_mode` is set from `mode` in in `BrowseLibrary.pm` (sub `_artists`). It's 'artists' when coming from a standard artists menu, otherwise it's something else.

I've updated the PR to cover the second point.